### ISSUE TITLE
Possible leak in SingleChronicleQueue::cleanupStoreFilesWithNoData()

### ISSUE THREAD

**Issue by garikjan** [OpenHFT/Chronicle-Queue#906]
Could you please check my findings: while iterating stores of previous cycles, stores are acquired, but not released. Additionally, it leads to unbalanced calls to onAcquire()/onRelease(). It was discovered in 5.21ea44, but is still actual for HEAD/master

Best regards,
Garik.

**Comment by garikjan** [comment]
Just a kind reminder. Still waiting for some feedback ;)

**Comment by JerryShea** [comment]
Hi @garikjan, we are pretty busy with paying work right now. There are already unit tests for this functionality e.g. AcquireReleaseTest, so it would be great if you could add a failing test (and even a fix!) to that and send us a PR.

If you would like to discuss commercial support please email sales@chronicle.software

**Comment by garikjan** [comment]
Hi @JerryShea, sorry for late response. I discovered, that is was not a leak. Resources are freed indeed. But anyway onReleased() is not called, as soon as method SingleChronicleQueueStore::close() is called directly by try-with-resources in SingleChronicleQueue::cleanupStoreFilesWithNoData() and onReleased() callback is not executed.

I created a PR, as you suggested: https://github.com/OpenHFT/Chronicle-Queue/pull/958
